AlembicImporter/Assets/UnitTests/AlembicExportTest.cs

```diff
@@ -0,0 +1,235 @@
+            }
+
+            // repeat until you find a file that does not already exist
+            do
+            {
+                temp = Path.Combine(pathName, MakeFileName(prefixName: prefixName, extName: extName));
+            } while (File.Exists(temp));
+
+            // Unity asset paths need a slash on all platforms.
+            if (unityPathSeparator)
+            {
+                temp = temp.Replace('\\', '/');
+            }
+
+            return temp;
+        }
+
+        /// <summary>
+        /// Return a random .abc path that you can use in
+        /// the File APIs.
+        /// </summary>
+        protected string GetRandomAbcFilePath()
+        {
+            return GetRandomFileNamePath(extName: ".abc", unityPathSeparator: false);
+        }
+        // -------------------------------------------------
+
+
+        [TearDown]
+        public virtual void Term()
+        {
+            if (string.IsNullOrEmpty(_testDirectory))
+            {
+                return;
+            }
+
+            // Delete the directory on the next editor update.  Otherwise,
+            // prefabs don't get deleted and the directory delete fails.
+            EditorApplication.update += DeleteOnNextUpdate;
+        }
+
+        // Helper for the tear-down. This is run from the editor's update loop.
+        void DeleteOnNextUpdate()
+        {
+            EditorApplication.update -= DeleteOnNextUpdate;
+            try
+            {
+                Directory.Delete(filePath, recursive: true);
+                AssetDatabase.Refresh();
+            }
+            catch (IOException)
+            {
+                // ignore -- something else must have deleted this.
+            }
+        }
+    }
+
+    public class AlembicExportTest : AlembicTestBase
+    {
+        private const string c_scene = "TestCloth";
+
+        private AlembicExporter GetAlembicExporter()
+        {
+            var alembicExporter = Object.FindObjectOfType<AlembicExporter>();
+            Assert.That(alembicExporter, Is.Not.Null);
+            return alembicExporter;
+        }
+
+        private string GetAssetsAbsolutePath(string relPath)
+        {
+            return Application.dataPath + relPath.Replace("Assets", "");
+        }
+
+        /// <summary>
+        /// Test that the abc file was reimported and can be loaded into Unity through
+        /// the asset database.
+        /// </summary>
+        /// <param name="abcPath"></param>
+        private void TestAbcImported(string abcPath)
+        {
+            AssetDatabase.Refresh();
+
+            var absPath = GetAssetsAbsolutePath(abcPath);
+
+            Assert.That(string.IsNullOrEmpty(absPath), Is.False);
+            Assert.That(File.Exists(absPath));
+
+            // now try loading the asset to see if it imported properly
+            var obj = AssetDatabase.LoadMainAssetAtPath(abcPath);
+            Assert.That(obj, Is.Not.Null);
+            var go = obj as GameObject;
```

Comment:
Is there a way to dig into the GameObject a little more to confirm it is coming from an Alembic file? Look for the "streaming" component for example?
Something  Alexis can take care of in a future pull request

Reply:
Idea: for regular exports (not one-shot) we could also test for the number of frames through the max value for the time property

Reply:
good ideas!